packages/vite/src/node/plugins/css.ts

```diff
@@ -120,6 +120,8 @@ export const chunkToEmittedCssFileMap = new WeakMap<
   Set<string>
 >()
 
+export const removedPureCssFiles = new Map<string, RenderedChunk>()
```

Comment:
We could have issues with this `Map` in build watch mode. @crcong would you check how other caches like `cssModulesCache` are implemented so we avoid these problems. You'll need to make the map depend on the resolved config and re-initialize it in `buildStart` that is called at the beginning of each re-build.

Reply:
I didn't notice that. Thanks for the detailed explanation. I've changed it. Please review my code again when you have free time.